client/landing/gutenboarding/domain-suggestions/index.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import { controls } from '@wordpress/data-controls';
+import { registerStore } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import reducer, { State } from './reducer';
+import * as actions from './actions';
+import * as selectors from './selectors';
+import * as resolvers from './resolvers';
+import { TailParameters } from '../store/types';
+
+export { STORE_KEY };
+
+registerStore< State >( STORE_KEY, {
+	actions,
+	controls,
+	reducer,
+	resolvers,
+	selectors,
+	// persist: [],
+} );
+
+type Select = {
+	[ selector in keyof typeof selectors ]: (
+		...args: TailParameters< typeof selectors[ selector ] >
+	) => ReturnType< typeof selectors[ selector ] >;
+};
+
+type Dispatch = {
+	[ A in keyof typeof actions ]: ( ...args: Parameters< typeof actions[ A ] > ) => void;
+};
```

Comment:
These seemed a bit trickier since `selectors` and `actions` are actual (imported) symbols here, so I think I'd need to do some type arithmetics to make `Select` and `Dispatch` type 'functions' of type 'arguments'. I guess that's possible; I just haven't done it before.

But now that I look at them again, they're probably not so different from `TailParameters` :thinking:

Reply:
Okay, it is kinda tricky :grimacing: I'll defer this to later (I'll file an issue).

Reply:
I took this on in https://github.com/Automattic/wp-calypso/pull/37511